File: Golang/383. Ransom Note.go
func canConstruct(ransomNote string, magazine string) bool {
    charDict := [26]int{}
    for i := range magazine {
        charDict[int(magazine[i]) - 'a']++
    }
    for i := range ransomNote {
        charDict[int(ransomNote[i]) - 'a']--
    }
    for i := range charDict {
        if charDict[i] < 0 {
            return false
        }
    }
    return true
}

File: Python3/383. Ransom Note.py
class Solution:
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        dict1 = collections.Counter(ransomNote)
        dict2 = collections.Counter(magazine)
        dict2.subtract(dict1)
        for value in dict2.values():
            if value < 0:
                return False
        return True
